### ISSUE TITLE
Implement a deletion job

### ISSUE THREAD

**Issue by jsanda** [k8ssandra/k8ssandra#60]
**Is your feature request related to a problem? Please describe.**
As mentioned in #59 deletions in Helm are unordered. If we consolidate the k8ssandra and k8ssandra-cluster charts, we need to ensure that the CassandraDatacenter is deleted before cass-operator. 

**Describe the solution you'd like**
We can declare a `pre-delete` hook on the chart. The hook will be configured to run a `Job` that performs ordered deletions of resources. When the user executes `helm uninstall`, Helm will wait to run until the deletion job completes successfully. Really the `Job` only needs to worry about deleting the CassandraDatacenter and any other objects with finalizers for that matter. We can let Helm take care of everything else.

In summary we will need the following:

* A k8s client application with the required deletion logic
    * We will write this in Go
    * We will use the controller-runtime dynamic client
* A Docker image of the application
* A k8s `Job` that runs the image

@jeffbanks I am going to assign this to you since you have been doing related work in #8. I can work with you to get things set up and configured.

**Comment by jeffbanks** [comment]
Appears that this will become resolved with the modifications for merging to a single chart where the deletion job for uninstall is being added as well.

Referencing the PR [here](https://github.com/k8ssandra/k8ssandra/pull/116)

**Comment by jeffbanks** [comment]
@burmanm assigned this to you so you get credit for this one.  Looks like you are addressing with your other ticket.
